scripts/release-autoscaler.sh

```diff
@@ -0,0 +1,244 @@
+   echo " - bumping version to ${version} in API files at revision $(git rev-parse HEAD)"
+   yq eval -i ".build = \"${version}\"" api/default_info.json
+
+   git add api/default_info.json
+   git commit -S -m "Updated release version to ${version} in golangapiserver"
+}
+
+function create_mtar() {
+  set -e
+  mkdir -p "${build_path}/artifacts"
+  local version=$1
+  local build_path=$2
+  echo " - creating autorscaler mtar artifact"
+  pushd "${autoscaler_dir}" > /dev/null
+    make mta-release VERSION="${version}" DEST="${build_path}/artifacts/"
+  popd > /dev/null
+}
+
+function create_tests() {
+  set -e
+  mkdir -p "${build_path}/artifacts"
+  local version=$1
+  local build_path=$2
+  echo " - creating acceptance test artifact"
+  pushd "${autoscaler_dir}" > /dev/null
+    make acceptance-release VERSION="${version}" DEST="${build_path}/artifacts/"
+  popd > /dev/null
+}
+
+function commit_release(){
+  pushd "${autoscaler_dir}"
+  git add -A
+  git status
+  git commit -S -m "created release v${VERSION}"
+}
+
+function determine_next_version(){
+  echo " - Determining next version..."
+  echo " - Previous version: $previous_version"
+
+  # Remove 'v' prefix if present
+  local version_number=${previous_version#v}
+
+  # Parse version components
+  IFS='.' read -r major minor patch <<< "$version_number"
+
+  # Get commits since last tag
+  local commits_since_tag
+  commits_since_tag=$(git rev-list "${previous_version}"..HEAD --oneline 2>/dev/null || git rev-list HEAD --oneline)
+  local commit_count
+  commit_count=$(echo "$commits_since_tag" | wc -l)
+
+  if [ -z "$commits_since_tag" ] || [ "$commit_count" -eq 0 ]; then
+    echo " - No commits since last tag, keeping current version"
+    echo "$version_number" > "${build_path}/name"
+    return
+  fi
+
+  # Extract PR numbers from commits (supports both "(#123)" and " #123 " formats)
+  local pr_numbers
+  pr_numbers=$(echo "$commits_since_tag" | grep -oE '(\(#[0-9]+\)| #[0-9]+ )' | grep -oE '[0-9]+' | sort -u)
+
+  if [ -z "$pr_numbers" ]; then
+    echo " - No PR numbers found in commits, incrementing patch version"
+    patch=$((patch + 1))
+    local new_version="${major}.${minor}.${patch}"
+    echo " - Next version: $new_version"
+    echo "$new_version" > "${build_path}/name"
+    return
+  fi
+
+  # Query GitHub API for PR labels and categorize
+  local has_breaking=0
+  local has_enhancement=0
+  local pr_count=0
+
+  echo " - Checking PR labels for version determination..."
+  while IFS= read -r pr_num; do
+    if [ -n "$pr_num" ]; then
+      pr_count=$((pr_count + 1))
+      local labels
+      labels=$(gh pr view "$pr_num" --json labels --jq '.labels[].name' 2>/dev/null || echo "")
+
+      if echo "$labels" | grep -q "exclude-from-changelog"; then
+        echo "   - PR #$pr_num: excluded from changelog"
+        continue
+      fi
+
+      if echo "$labels" | grep -q "breaking-change"; then
+        echo "   - PR #$pr_num: breaking change"
+        has_breaking=1
```

Comment:
Just thinking about the ratelimits we have encountered: Here we could already stop looking at further PRs.

But it would only help in very few cases